File: Dataset/Leetcode/train/75/564.java
 class Solution {
    public void XXX(int[] nums) {
        int start = 0;
        int last = nums.length - 1;
        int temp = 0;
        while(temp != last){
            if(nums[temp] == 2){
                while(nums[last] == 2){
                    if(temp == last) return ;
                    last--;
                }
                swap(nums,temp,last);
            }else if(nums[temp] == 0){
                while(nums[start] == 0){
                    if(start == last) return ;
                    start++;
                }
                if(start >= temp) temp = start;
                else{
                    swap(nums,start,temp);
                }
            }else{
                if(nums[last] == 0) swap(nums,last,temp);
                else temp++;
            }
        }
    }
    public void swap(int[] nums , int i , int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: Dataset/Leetcode/test/55/564.cpp
 class Solution {
public:
    
    bool XXX(vector<int>& nums) {
        //避开0
        //每次尽可能跳跃长的长度？不对
        //从后往前找，遇到0就在0之前找有没有能跳过这个0的步数，找到起点都没找到就返回f
        //找到了就从该数之前开始继续查找0，直到找到起点
        if (nums.size() == 1 && nums[0] == 0)
            return true;
        for(int i=nums.size()-1;i>=0;--i)
        {
            if(nums[i]==0)
            {
                int j=i-1;
                for(;j>=0;--j)
                {
                    if(nums[j]>i-j|| (i == nums.size() - 1 && nums[j] >= i - j))//找到能够跳过0的长度
                    {
                        i=j;
                        break;
                    }
                }
                if(j==-1)//没有找到
                return false;
            }
        }
        return true;   
    }
};

